pkg/ddevapp/ddevapp.go

```diff
@@ -569,18 +572,16 @@ func (app *DdevApp) GetPrimaryRouterHTTPPort() string {
 	if app.RouterHTTPPort != "" {
 		proposedPrimaryRouterHTTPPort = app.RouterHTTPPort
 	}
-	if httpExpose := app.GetWebEnvVar("HTTP_EXPOSE"); httpExpose != "" {
-		//util.Debug("GetPrimaryRouterHTTPPort(): HTTP_EXPOSE='%s'", httpExpose)
-		httpPort := app.RouterPortFromExposeVariable(httpExpose, proposedPrimaryRouterHTTPPort)
-		if httpPort == "" {
-			//util.Debug("GetPrimaryRouterHTTPPort(): no primary router port found in %s", httpExpose)
-			proposedPrimaryRouterHTTPPort = ""
-		}
+	if httpPort := app.GetWebEnvVar("DDEV_ROUTER_HTTP_PORT"); httpPort != "" {
```

Comment:
Why use this instead of app.GetPrimaryRouterHTTPPort() ? It seems like we're going in circles on that.

Reply:
I don't understand.
We are inside the `app.GetPrimaryRouterHTTPPort()`, if I call it here, it will be an endless loop.

Reply:
This variable `DDEV_ROUTER_HTTP_PORT` contains the used real port (either from config or ephemeral).

Reply:
Right, but DDEV_ROUTER_HTTP_PORT is set to the value of app.GetPrimaryRouterHTTPPort(), so that what I mean by circular.

Reply:
Now I understand what you mean, `DDEV_ROUTER_HTTP_PORT` is set before `app.ComposeYaml` is populated (`GetWebEnvVar` reads `app.ComposeYaml`), so there is no issue.

This has no difference with what we use here before , `HTTP_EXPOSE` was handled in the same way. I just found that we don't need to read `HTTP_EXPOSE`, if it has it's own variable.